File: frontend/src/types/banner.ts
export enum BannerGroup {
  MAIN_A = '메인A',
  MAIN_B = '메인B',
  SUB_A = '서브A',
  SUB_B = '서브B',
  CORP_AD = '기업광고'
}

export enum BannerStatus {
  ACTIVE = 'ACTIVE',
  INACTIVE = 'INACTIVE'
}

export interface Banner {
  id: number;
  groupName: BannerGroup;
  title: string;
  imageUrl: string;
  linkTarget: string;
  linkTargetType: '_blank' | '_self';
  startDate: string;
  endDate: string;
  status: BannerStatus;
} 

File: backend/src/main/kotlin/com/yourjob/backend/disabled-packages/entity/banner/Banner.kt
package com.yourjob.backend.entity.banner

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "banners")
data class Banner(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    @Column(name = "banner_id", nullable = false, length = 100)
    val bannerId: String,

    @Column(name = "created_at", nullable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "status", nullable = false)
    var status: String,

    @Column(name = "updated_at", nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now(),

    @Column(name = "start_date", nullable = false, length = 50)
    var startDate: String,

    @Column(name = "end_date", nullable = false, length = 50)
    var endDate: String,

    @Column(name = "image_url", nullable = false, length = 255)
    var imageUrl: String,

    @Column(name = "group_name", nullable = false, length = 100)
    var groupName: String,

    @Column(name = "title", nullable = false, length = 255)
    var title: String,

    @Column(name = "link_target", nullable = true, length = 255)
    var linkTarget: String? = null,

    @Column(name = "link_target_type", nullable = true, length = 50)
    var linkTargetType: String? = null
)